R/obsPlot.R

```diff
@@ -1,274 +1,453 @@
+                                      prior.count = convert_prior.count) %>%
+            as.data.frame()
+    } else {
+        data <- getItem(DGEdata, countsMatrix) %>%
+            as.data.frame()
     }
 
-    assertthat::assert_that(!missing(data),
-                            "data.frame" %in% class(data),
-                            msg = "data must be specified and should be of class 'data.frame'.")
-    assertthat::assert_that(plotByCol %in% colnames(data),
-                            msg = 'The plotByCol must be included in the colnames of the specified data.')
-    assertthat::assert_that(groupCol %in% colnames(data),
-                            msg = "The groupCol must be included in the colnames of the specified data.")
-    assertthat::assert_that(valueCol %in% colnames(data),
-                            msg = "The valueCol must be included in the colnames of the specified data.")
-    assertthat::assert_that(all(groupOrder %in% as.character(data[groupCol, , drop = TRUE])))
-
-    # Reduce box outliers to a dot if geom_points turned on
-    outlier.size <- 1.5
-    outlier.shape <- 19
-    if (pointLayer) {
-        outlier.size <- 1
-        outlier.shape <- "."
+        if (any(is.null(designTable),
+                !is.character(designTable),
+                !(designTable %in% names(DGEdata)))) {
+            if ("design" %in% names(DGEdata)) {
+                designTable <- "design"
+                warning("designTable specified is not present in DGEobj. Assigning default value of type 'design'.")
+            } else if (!is.null(getType(DGEdata,"design"))) {
+                designTable <- names(getType(DGEdata,"design"))
+            }
+        }
+
+    design            <- DGEobj::getItem(DGEdata, designTable)
+    colnames(design)  <- tolower(colnames(design))
+    group_default     <- NULL
+    if (tolower("ReplicateGroup") %in% colnames(design)) {
+        group_default <- "replicategroup"
     }
 
-    # Plot code here
-    if (facet == TRUE) {
-        # Set facet columns to sqrt of unique observations (rounded up)
-        if (missing(facetRow)) {
-            numcol <- data[plotByCol] %>% unique %>% length %>% sqrt %>% ceiling
+    if (any(is.null(group),
+            !(tolower(group) %in% colnames(design)))) {
+        if (!is.null(group_default)) {
+            warning("group must be specified and should be one of the columns in the design object in DGEdata. Assigning ",
+                    group_default,
+                    " as the default value.")
+            group <- group_default
         } else {
-            numcol <- facetRow
+            stop("group must be specified and should be one of the columns in the designTable in DGEdata.")
         }
+    }
+
+    # Create a rownames column
+    data$GeneID    <- rownames(data)
+    rownames(data) <- NULL
+    data           <- tidyr::gather(data, key = "sample", value = "value",-GeneID)
```

Comment:
`tidyr::gather `is not in **importFrom**

Reply:
present in import from